core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -51,6 +58,7 @@ class ControllerContext(val zkUtils: ZkUtils) {
   var partitionLeadershipInfo: mutable.Map[TopicAndPartition, LeaderIsrAndControllerEpoch] = mutable.Map.empty
   val partitionsBeingReassigned: mutable.Map[TopicAndPartition, ReassignedPartitionsContext] = new mutable.HashMap
   val partitionsUndergoingPreferredReplicaElection: mutable.Set[TopicAndPartition] = new mutable.HashSet
+  val replicasOnOfflineDisks: mutable.Map[Int, Set[TopicPartition]] = mutable.HashMap.empty
```

Comment:
Yeah, we definitely want to migrate all the code to the Java class. It probably makes sense to do the whole class in one go though.